index.js

```diff
@@ -138,6 +148,266 @@ app.intent('send.message - no', async conv => {
   conv.contexts.set('anything_else', 2);
 });
 
+/**
+ * join.conference
+ */
+app.intent('join.conference', async (conv, {target}) => {
+  const circuit = await getCircuit(conv);
+  if(!circuit){
+    return;
+  }
+
+  let matches = [];
+  let { inConferenceCall, conferenceAvailable, calls } = await getConferenceInfo(conv);
+
+  if (inConferenceCall) {
+    conv.ask("<speak>You're already in a conference call. Can I do anything else for you?</speak>");
+    conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+    conv.contexts.set('anything_else', 2); 
+    return;
+  } 
+
+  if(!conferenceAvailable){
+    conv.ask(`<speak>No conferences available to join. Can I do something else for you?</speak>`);
+    conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+    conv.contexts.set('anything_else', 2);
+    return;
+  }
+
+  // If target isn't undefined
+  if(target){
+    calls.forEach(c => {
+      if(c.title.toLowerCase() === target.toLowerCase()){
+        matches.push(c);
+      }
+    });
+
+    if (!matches.length) {
+      let prompt = `<speak>I cannot find any conference call with name ${target}. Here are the names of your current ongoing conferences. 
+      Which would you like to join?</speak>`;
+      let titles = calls.map((call) => call.title.toLowerCase());
+      conv.ask(prompt, new Suggestions(titles));
+      conv.contexts.set('joinconference_gettarget', 5, {
+        calls: calls
+      });
+      return;
+    }
+
+    if (matches.length === 1) {
+      // One result found. Ask user for confirmation.
+      const confId = matches[0].callId;
```

Comment:
There isn't really such a thing as a conference ID. There are conversation IDs and call IDs. This is call ID. Use "callId" as variables since everyone will understand this.